UI Fixes: History View Refinements

sachasayan opened the pull request:
## Context

![Screenshot 2025-04-20 at 11 56 16 PM](https://github.com/user-attachments/assets/45f87c41-28c2-4284-827e-5890c0ff67ca)


- General UX/UI Refinements on History View: Improved padding, improved iconography.
- Dropdown for sorting instead of radio buttons. 
- Allow selectable workspaces instead of just "all" workspaces. 
- Modifications primarily to HistoryView.tsx, but this PR touches a few other files in mostly minor ways. 
- Notably, `useTaskSearch` was upgraded to expand filtering capabilities. 
- **All functionality is preserved, just improved or re-styled. :)** 

![Screenshot 2025-04-20 at 11 58 08 PM](https://github.com/user-attachments/assets/b75a7e82-78e2-41b9-9453-d63121400dee)


## Video


https://github.com/user-attachments/assets/9ffcfbc1-d4c3-4f12-8bd5-4162abda02e7



## Get in Touch

As always, InElement on Discord.

sachasayan commented:
Draft btw, do not merge. ✌️

elianiva commented:
hmm i'm not sure if it's good to remove selection mode and replace it with how it works in this PR, i'm expecting when you click on a task then it will open it

clicking while holding shift is more common i think, but to make users discover that without being too ambiguous or add another clutter like "hold shift to select" is something that needs figuring out

maybe keep the selection mode but automatically triggers it whenever one of the task is selected? no need to click "selection mode" manually
so you'd have a checkbox like how it was before—hidden by default, just like the other icons—but once the users click that, the entire card will be selectable, best of both worlds

sachasayan commented:
@elianiva Played around with this last night and tried a checkbox-based variant.... I didn't like it as much. Click target is too small, and/or you end up taking up more horizontal space. Gave it an honest shot, but the vibes weren't there.  I'll push another commit in a bit (gotta fix some tests) but as a preview of what I've done to address some of the concerns: 

1. Action icons (trash, copy, export) don't go from opacity-0 to opacity-100 on hover. Instead, they go from opacity-50 to opacity-100. This means they're visible but they don't add too much visual repetition in the default state. It seems to work well as a compromise to me, I hope you agree. 
2. Switched to single select items by default. This makes them act more like a typical file system explorer. Click on, click off, and ctrl-click and shift-click if you want to select more items. 
3. Added more real estate to the navigate button. I think making the whole component a navigation button is a UX mistake tbh, but I agree you want the hit target on the larger size.

sachasayan commented:
This one's still here, ready to go.

mrubens commented:
For what it's worth my intuition is also that clicking on the history row should take you to the task, similar to how it works in the history preview.

mrubens commented:
I think the rest of these changes are good, would just vote to bring back the selection as a secondary action instead of the primary action.

daniel-lxs commented:
I'll close this PR as stale, if someone else wants to work on this they can open a new PR.
